WellLog.Lib/Models/DLIS/ComponentDescriptor.cs

```diff
@@ -0,0 +1,50 @@
+﻿using WellLog.Lib.Helpers;
+
+namespace WellLog.Lib.Models.DLIS
+{
+    public class ComponentDescriptor
+    {
+        private const byte ABSENT_ATTRIBUTE_ROLE = 0b_0000_0000;
+        private const byte ATTRIBUTE_ROLE = 0b_0010_0000;
+        private const byte INVARIANT_ATTRIBUTE_ROLE = 0b_0100_0000;
+        private const byte OBJECT_ROLE = 0b_0110_0000;
+        private const byte REDUNDANT_SET_ROLE = 0b_1010_0000;
+        private const byte REPLACEMENT_SET_ROLE = 0b_1100_0000;
+        private const byte SET_ROLE = 0b_1110_0000;
+
+        private const byte SET_TYPE_MASK = 0b_0001_0000;
+        private const byte SET_NAME_MASK = 0b_0000_1000;
+
+        private const byte OBJ_NAME_MASK = 0b_0001_0000;
+
+        private const byte ATTR_LABEL_MASK = 0b_0001_0000;
+        private const byte ATTR_COUNT_MASK = 0b_0001_0000;
+        private const byte ATTR_REPRESENTATION_CODE_MASK = 0b_0001_0000;
+        private const byte ATTR_UNITS_MASK = 0b_0001_0000;
+        private const byte ATTR_VALUE_MASK = 0b_0001_0000;
+
+        private readonly byte _descriptor;
+
+        public bool IsAbsentAttribute => _descriptor.IsComponentRole(ABSENT_ATTRIBUTE_ROLE);
+        public bool IsAttribute => _descriptor.IsComponentRole(ATTRIBUTE_ROLE);
+        public bool IsInvariantAttribute => _descriptor.IsComponentRole(INVARIANT_ATTRIBUTE_ROLE);
+        public bool IsObject => _descriptor.IsComponentRole(OBJECT_ROLE);
+        public bool IsRedundantSet => _descriptor.IsComponentRole(REDUNDANT_SET_ROLE);
+        public bool IsReplacementSet => _descriptor.IsComponentRole(REPLACEMENT_SET_ROLE);
+        public bool IsSet => _descriptor.IsComponentRole(SET_ROLE);
+
+        public bool SetHasType => (IsSet || IsRedundantSet || IsReplacementSet) ? _descriptor.GetBit(SET_TYPE_MASK) : false;
+        public bool SetHasName => (IsSet || IsRedundantSet || IsReplacementSet) ? _descriptor.GetBit(SET_NAME_MASK) : false;
```

Comment:
Maybe it's been too long since I've used C# but it doesn't look like these are modified anything? Get rid of the word "Set" within these names.

Reply:
the word `Set` in this context refers to a [SetComponent](http://w3.energistics.org/rp66/v1/rp66v1_sec3.html#3_2_2_2) rather than setting a value.  do you even read requirements, bruh?

Reply:
Sounds like something that might be better named. Set can refer to a collection, action on an object, or what looks like some domain term in this log file.

How about:

`DoesSetComponentHaveType`
`DoesSetComponentHaveName`

Reply:
agreed...done